SAMtags.tex

```diff
@@ -453,6 +453,165 @@ \subsubsection{Color space}
+% The qualities are encoded in the same manner as the primary {\sf QUAL} field; one byte per quality with ASCII value Phred score + 33.
+% A space character (`{\tt \textvisiblespace}') should be used as a separator between concatenated quality strings when multiple modification lists are present in the {\tt MM} tag.
+% The length should match the number of position deltas from {\tt MM} plus 1 per space character required.
+%
+% For example ``{\tt MM:Z:C+m,5,12,3;C+h,57;}'' may have an associated
+% quality tag of ``{\tt MP:Z:5EB /}''.
+%
+% Where multiple modification types are listed together, such as in ``{\tt MM:Z:C+mh,5,12,3;}'' the quality values are interleaved in order ({\tt m} at 6, {\tt h} at 6, {\tt m} at 19, {\tt h} at 19 and so on), giving 6 quality values in total for this example.
+%
+% Quality values for ambiguity codes give the likelihood that the
+% modification is one of the possible codes compatible with that
+% ambiguity code.  For example {\tt MM:Z:C+C,10; MP:Z:+} indicates a C
+% call with an unspecified modification and the phred score of 10 (ASCII
+% value {\tt +}).  This corresponds to a 90\% chance of the base being
+% modified.
+%
+% To represent several possible modifications at the same site the {\tt MP} tag can be used to indicate the probabilities of each possibility.
+% The values used should be absolute probabilities, not relative between the alternatives.
+% For example, a C base that has 95\% chance of being modified with 5mC being three times more likely than 5hmC will encode 5mC with 67.5\% probability ($0.9 * 0.75$ giving phred score 5, ASCII value {\tt \&})and 5hmC with 22.5\% probability ($0.9 * 0.25$ giving phred score 1, ASCII value {\tt "}).
+% This could be represented with ``{\tt MM:Z:C+m,10;C+h,10; MP:Z:" \&}''.
+
+\item[Ml:B:C,\tagvalue{scaled-probabilities}]
+\hfill\\
+The optional {\tt Ml} tag lists the probability of each modification listed in the {\tt Mm} tag being correct, in the order that they occur.
+The continuous probability range 0.0 to 1.0 is remapped in equal
+sized portions to the discrete integers 0 to 255 inclusively. Thus the
+probability range corresponding to integer value $N$ is $N/256$ to
+$(N+1)/256$.
+
+The SAM encoding therefore uses a byte array of type `{\tt C}' with the number of elements matching the summation of the number of modifications listed as being present in the {\tt Mm} tag accounting for multi-modifications each having their own probability.
+
+For example ``{\tt Mm:Z:C+m,5,12;C+h,5,12;}'' may have an associated tag of ``{\tt Ml:B:C,204,89,26,130}''.
+
+If the above is rewritten in the multiple-modification form, the probabilities are interleaved in the order presented, giving ``{\tt Mm:Z:C+mh,5,12;  Ml:B:C,204,26,89,130}''.
+Note where several possibile modifications are presented at the same site, the {\tt Ml} values represent the absolute probabilities of the modification call being correct and not the relative likelihood between the alternatives.
```

Comment:
possibile->possible

Reply:
if the Ml is _likelihood_ then we shouldn't confuse further by saying that it's the "absolute _probability_"....what it should be is the likelihood of making the measurement that was made assuming the modification is as stated. i.e P(data | modification) not P(modification|data).

Reply:
to get  to a posterior probablity one would need a prior...

Reply:
I was fearful of using a phrase like likelihood as it can be interpreted either in a very specific statistical way or in a more casual English speaking way.  I used `l` in the tag name because I'd previously used `p` for the phred score and `MQ` tag already exists.  Maybe that's wrong and I should have gone with `MP` for this linear scale and if we ever switch use `MZ`, as we sometimes use Z already; eg barcode phred scores.

I admit my stats is pretty ropey though so I'm not really sure exaclty what it should be.  Maybe @marcus1487 can advise?  My instinct though is that for phred base calls we're describing the probability of the observed call being the real one, so that's P(observed | real)?  I think that agrees with your P(data | modification) statement.

As for priors, if I recall that's exactly what I was doing in the `samtools consensus` PR using a prior likelihood of modifications, but I'd have to do some hunting to see if that was a plan or whether I put it into fruition. :)